src/main/java/ru/practicum/shareit/booking/BookingServiceImpl.java

```diff
@@ -0,0 +1,190 @@
+        List<Booking> bookings;
+
+        switch (bookingState) {
+            case ALL:
+                bookings = repository.findAllByBooker(userId, sort);
+                break;
+            case CURRENT:
+                bookings = repository.findByBookerAndStartIsBeforeAndEndIsAfter(userId, dateNow, dateNow, sort);
+                break;
+            case PAST:
+                bookings = repository.findByBookerAndEndIsBefore(userId, dateNow, sort);
+                break;
+            case FUTURE:
+                bookings = repository.findByBookerAndStartIsAfter(userId, dateNow, sort);
+                break;
+            case WAITING:
+                bookings = repository.findByBookerAndStartIsAfterAndStatusIs(userId, dateNow, sort, BookingStatusEnum.WAITING);
+                break;
+            case REJECTED:
+                bookings = repository.findByBookerAndStartIsAfterAndStatusIs(userId, dateNow, sort, BookingStatusEnum.REJECTED);
+                break;
+            default:
+                return List.of();
+        }
+        List<BookingDtoResponse> bookingDtoResponse = new ArrayList<>();
+        for (Booking bok : bookings) {
+            bookingDtoResponse.add(BookingMapper.toBookingDto(bok, userService.get(bok.getBooker()), itemService.get(bok.getItemId())));
+        }
+        return bookingDtoResponse;
+    }
+
+    @Override
+    public List<BookingDtoResponse> getAllByOwner(Long ownerId, String state) {
+        UserDto userDto = userService.get(ownerId);
+        LocalDateTime dateNow = LocalDateTime.now();
+        Sort sort = Sort.by("start").descending();
+        State bookingState = State.stringToState(state)
+                .orElseThrow(() -> new InvalidStatusException("Unknown state: " + state));
+        List<Long> itemIdList = itemService.getItems(userDto.getId())
+                .stream()
+                .map(ItemDtoForBooking::getId)
+                .collect(Collectors.toList());
+
+        List<Booking> bookings;
+        switch (bookingState) {
+            case ALL:
+                bookings = repository.findAllByItemIdIn(itemIdList, sort);
+                break;
+            case CURRENT:
+                bookings = repository.findByItemIdInAndStartIsBeforeAndEndIsAfter(itemIdList, dateNow, dateNow, sort);
+                break;
+            case PAST:
+                bookings = repository.findByItemIdInAndEndIsBefore(itemIdList, dateNow, sort);
+                break;
+            case FUTURE:
+                bookings = repository.findByItemIdInAndStartIsAfter(itemIdList, dateNow, sort);
+                break;
+            case WAITING:
+                bookings = repository.findByItemIdInAndStartIsAfterAndStatusIs(itemIdList, dateNow, sort, BookingStatusEnum.WAITING);
+                break;
+            case REJECTED:
+                bookings = repository.findByItemIdInAndStartIsAfterAndStatusIs(itemIdList, dateNow, sort, BookingStatusEnum.REJECTED);
```

Comment:
@Query(" select b " +
			"from Booking b " +
			"where b.item.owner.id = ?1 " +
			"  and (b.status = 'CANCELED' or b.status = 'REJECTED' )")
	List<Booking> findOwnerCanceled(Long bookerId, Sort sort);